File: remote/src/main/java/com/kelsos/mbrc/presenters/PlaylistTrackPresenter.java
package com.kelsos.mbrc.presenters;

import com.kelsos.mbrc.annotations.Queue;
import com.kelsos.mbrc.domain.PlaylistTrack;
import com.kelsos.mbrc.ui.views.PlaylistTrackView;

public interface PlaylistTrackPresenter {
  void bind(PlaylistTrackView view);

  void load(long longExtra);

  void queue(PlaylistTrack track, @Queue.Action String action);
}

File: app/src/main/kotlin/com/kelsos/mbrc/presenters/PlaylistTrackPresenter.kt
package com.kelsos.mbrc.presenters

import com.kelsos.mbrc.annotations.Queue
import com.kelsos.mbrc.domain.PlaylistTrack
import com.kelsos.mbrc.ui.views.PlaylistTrackView

interface PlaylistTrackPresenter {
  fun bind(view: PlaylistTrackView)

  fun load(longExtra: Long)

  fun queue(track: PlaylistTrack, @Queue.Action action: String)
}
